docs/admin/k8s-deployment.md

```diff
@@ -1,172 +1,404 @@
+specific errors back to the user. K8s users/controllers are only allowed to view
+these objects, NOT create or edit them.  Below is an example ebpfProgram Object
+which was automatically generated in response to the above BpfProgramConfig
+Object.
+
+```yaml
+apiVersion: bpfd.io/v1alpha1
+  kind: BpfProgram
+  metadata:
+    creationTimestamp: "2022-12-07T22:41:29Z"
+    finalizers:
+    - bpfd.io.agent/finalizer
+    generation: 2
+    labels:
+      owningConfig: xdp-pass-all-nodes
+    name: xdp-pass-all-nodes-bpfd-deployment-worker2
+    ownerReferences:
+    - apiVersion: bpfd.io/v1alpha1
+      blockOwnerDeletion: true
+      controller: true
+      kind: BpfProgramConfig
+      name: xdp-pass-all-nodes
+      uid: 6e3f5851-97b1-4772-906b-3ac69c6a4057
+    resourceVersion: "1506"
+    uid: 384d3d5c-e62b-4be3-9bf0-c6cf0e315acf
+  spec:
+    programs:
+      bdeac6d3-4128-464e-9161-6010684eca27:
+        attachpoint:
+          interface: eth0
+        maps: {}
+  status:
+    conditions:
+    - lastTransitionTime: "2022-12-07T22:41:30Z"
+      message: Successfully loaded ebpfProgram
+      reason: bpfdLoaded
+      status: "True"
+      type: Loaded
+```
+
+Applications wishing to use bpfd to deploy/manage their bpf programs in
+kubernetes will make use of this object to find references to the bpfMap pin
+points (`spec.maps`) in order to configure their bpf programs.
+
+## Controllers
+
+The Bpfd-Operator performs a few major functions and houses two major
+controllers the `bpfd-agent` and `bpfd-operator`.
+
+### bpfd-agent
+
+The bpfd-agent controller is deployed alongside bpfd in a daemonset.  It's main
+purpose is to watch user intent (in BpfProgramConfig Objects) and communicate
+with bpfd via a mTLS secured connection in order to translate the cluster-wide
+user-intent to per node state.
+
+### bpfd-operator
+
+The bpfd-operator performs the following functionality:
+
+- Create and Reconcile the bpfd daemonset (including both the `bpfd` and
+  `bpfd-agent` processes) so that no manual edits can be completed.
+- Report cluster wide state back the the user with each BpfProgramConfig's
+  status field.
+
+## More useful commands
+
+1. Install Instances of Custom Resources:
+
+```sh
+kubectl apply -f config/samples/
+```
+
+2. Build and push your bpfd-agent and bpd-operator container images to the
+   location specified by `BPFD_AGENT_IMG` and `BPFD_OPERATOR_IMG`:
+
+```sh
+make build-images push-images BPFD_OPERATOR_IMG=<some-registry>/bpfd-operator:tag BPFD_AGENT_IMAGE=<some-registry>/bpfd-agent:tag
+```
+
+3. Deploy the operator and agent to a cluster with the image specified by
+   `BPFD_AGENT_IMG` and `BPFD_OPERATOR_IMG`:
+
+```sh
+make deploy BPFD_OPERATOR_IMG=<some-registry>/bpfd-operator:tag BPFD_AGENT_IMAGE=<some-registry>/bpfd-agent:tag
+```
+
+## Uninstall CRDs
+
+To delete the CRDs from the cluster:
+
+```sh
+make uninstall
+```
+
+## Undeploy controller
+
+UnDeploy the controller to the cluster:
```

Comment:
```suggestion
Undeploy the controller from a vanilla K8s Deployment cluster:
```
There is also a `make deploy` command. Not sure if you want to document it like the reinstall of the CRD or not.